pages/api/submit.tsx

```diff
@@ -128,14 +117,27 @@ const processFormData = async (
       return await setTimeout(() => res.status(200).json({ received: true }), 1000);
     }
 
-    // Add file S3 urls to payload once we start processing files through reliability queue
-    // For now just attach the file names
     for (const [key, value] of Object.entries(files)) {
       const fileOrArray = value;
       if (!Array.isArray(fileOrArray)) {
         if (fileOrArray.name) {
-          reqFields[key] = fileOrArray.name;
+          const result = await pushFileToS3(fileOrArray, fileOrArray.name);
+          if (result.isValid) reqFields[key] = result.successValue.location;
+          else throw new Error(result.errorReason);
         }
+      } else if (Array.isArray(fileOrArray)) {
+        fileOrArray.map(async (fileItem) => {
```

Comment:
I do this a lot by default as well but usually if we don't need a return value or array from the `map` call it can be switched to a `forEach` instead which returns type void.